contract/AElf.Contracts.MultiToken/TokenContract_Actions.cs

```diff
@@ -110,18 +129,70 @@ public override Empty Transfer(TransferInput input)
             return new Empty();
         }
 
+        #region Cross chain
+
+        public override Empty CrossChainCreateToken(CrossChainCreateTokenInput input)
+        {
+            var parentChainId = GetValidCrossChainContractReferenceState().GetParentChainId.Call(new Empty()).Value;
+            var tokenContractAddress = State.CrossChainTransferWhiteList[parentChainId];
+            Assert(tokenContractAddress != null, "Token contract address of parent chain not found.");
+            
+            var originalTransaction = Transaction.Parser.ParseFrom(input.TransactionBytes);
+
+            AssertCrossChainTransaction(originalTransaction, tokenContractAddress, nameof(Create));
+            
+            var originalTransactionId = originalTransaction.GetHash();
+            CrossChainVerify(originalTransactionId, input.ParentChainHeight, input.FromChainId, input.MerklePath);
+
+            CreateInput creationInput = CreateInput.Parser.ParseFrom(originalTransaction.Params);
+
+            RegisterTokenInfo(new TokenInfo
+            {
+                Symbol = creationInput.Symbol,
+                TokenName = creationInput.TokenName,
+                TotalSupply = creationInput.TotalSupply,
+                Decimals = creationInput.Decimals,
+                Issuer = creationInput.Issuer,
+                IsBurnable = creationInput.IsBurnable,
+                IssueChainId = parentChainId
+            });
+            return new Empty();
+        }
+
+        public override Empty RegisterCrossChainTokenContractAddress(RegisterCrossChainTokenContractAddressInput input)
+        {
+            var owner = GetOwnerAddress();
+            Assert(Context.Sender == owner, "No permission.");
+            
+            var originalTransaction = Transaction.Parser.ParseFrom(input.TransactionBytes);
+            AssertCrossChainTransaction(originalTransaction, Context.GetZeroSmartContractAddress(input.FromChainId),
+                nameof(ACS0Container.ACS0ReferenceState.ValidateSystemContractAddress));
+            
+            var validAddress = ExtractTokenContractAddress(originalTransaction.Params);
+            
+            var originalTransactionId = originalTransaction.GetHash();
+            CrossChainVerify(originalTransactionId, input.ParentChainHeight, input.FromChainId, input.MerklePath);
+
+            State.CrossChainTransferWhiteList[input.FromChainId] = validAddress;
+            
+            return new Empty();
+        }
+
         /// <summary>
-        /// Transfer token form a chain to another chain
+        /// Transfer token form this chain to another one.
         /// </summary>
         /// <param name="input"></param>
         /// <returns></returns>
         public override Empty CrossChainTransfer(CrossChainTransferInput input)
```

Comment:
When doing cross chain transfer, we reduce `Supply` in current chain by burning. When issuing more tokens, we check whether `Supply` is equal or lower than `TotalSupply`. In this case, how do we prevent issuer to issue without exceeding `TotalSupply`? (In a scenario where issuer issues new tokens then do a cross chain transfer and repeat doing this) Maybe we introduce one more variable `TotalIssued` to track the number of tokens issued?

`Supply`: Number of tokens in current chain (-/+ when doing cross chain transfer)
`TotalIssued`: Total number of tokens issued until now (+ every time new tokens issued on parent chain)
`TotalSupply`: Maximum number of tokens that can be issued

Reply:
@aerkol Yeah, we also noticed this issue and it is to be solved.